src/api/spec/db/data/adapt_review_and_duplicated_notifications_spec.rb

```diff
@@ -0,0 +1,76 @@
+require 'rails_helper'
+require Rails.root.join('db/data/20200702145811_adapt_review_and_duplicated_notifications.rb')
+
+RSpec.describe AdaptReviewAndDuplicatedNotifications, type: :migration do
+  describe 'up' do
+    subject { AdaptReviewAndDuplicatedNotifications.new.up }
+
+    let(:subscriber) { create(:confirmed_user) }
+    let(:bs_request_with_submit_action) { create(:bs_request_with_submit_action, state: :review) }
+    let(:subscriber_review) do
+      create(:user_review, by_user: subscriber, user_id: subscriber.id,
+                           bs_request: bs_request_with_submit_action)
+    end
+
+    context 'review notifiable' do
+      before do
+        create(:notification, :review_wanted, notifiable: subscriber_review, subscriber: subscriber)
+        subject
+      end
+
+      it 'associates review notifications to the corresponding bs_request' do
+        expect(Notification.find_by(subscriber: subscriber).notifiable).to eq(bs_request_with_submit_action)
+      end
+
+      it 'removes all review notifications' do
+        expect(Notification.where(notifiable_type: 'Review')).not_to exist
+      end
+    end
+
+    context 'comment and bs_request notifiable' do
+      let(:comment_request) { create(:comment_request, commentable: bs_request_with_submit_action) }
+      let(:comment_request02) { create(:comment_request, commentable: bs_request_with_submit_action) }
+
+      let(:first_request_notification) do
+        create(:notification, :request_state_change, notifiable: bs_request_with_submit_action,
+                                                     subscriber: subscriber, web: true)
+      end
+      let(:second_request_notification) do
+        create(:notification, :request_state_change, notifiable: bs_request_with_submit_action,
+                                                     subscriber: subscriber, web: true)
+      end
+      let(:first_comment_notification) do
+        create(:notification, :comment_for_request, notifiable: comment_request,
+                                                    subscriber: subscriber, web: true)
+      end
+      let(:second_comment_notification) do
+        create(:notification, :comment_for_request, notifiable: comment_request02,
+                                                    subscriber: subscriber, web: true)
+      end
+
+      before do
+        first_request_notification
+        second_request_notification
+        first_comment_notification
+        second_comment_notification
```

Comment:
I think we could use `let!` instead of calling them inside the `before` block

Reply:
I cannot use `let!` on `first_request_notification` and `first_comment_notification` because rubocop doesn't allow it since I don't refer to them in the `it` block's. But I need them to be in the database, and to create them in the right order with the `second` notifications. Mixing up `let!` and calling the other from the `before` block makes it hard to keep them in order and is hard to read IMO. That why I used the `let` for all of them and call them in order from the `before` block

Reply:
okey, thanks ^^

Reply:
I found the same issue in my PR, the only thing I did was to rename the `let`s so they were verbs and looked like an action (e.g. `create_first_request_notification`)